ohapi/utils_fs.py

```diff
@@ -168,6 +168,82 @@ def load_metadata_csv(input_filepath):
     return metadata
 
 
+def is_valid_datetime(date):
+    """
+    Check if creation date is in ISO1069 format.
+
+    :param date: This field is creation date of the file.
+    """
+    regex = (r'^(-?(?:[1-9][0-9]*)?[0-9]{4})-(1[0-2]|0[1-9])-' +
+             r'(3[01]|0[1-9]|[12][0-9])' +
+             r'T(2[0-3]|[01][0-9]):([0-5][0-9]):([0-5][0-9])(\.[0-9]+)' +
+             r'?(Z|[+-](?:2[0-3]|[01][0-9]):[0-5][0-9])?$')
+    match_iso8601 = re.compile(regex).match
+    if match_iso8601(date) is not None:
+            return True
+    else:
+        print("date is not in ISO1069 format")
+        return False
+
+
+def is_metadata_valid(file_metadata):
+    """
+    Check if metadata fields like description,tags and creation date are valid.
+
+    :param file_metadata: This field is metadata of file.
+    """
+    if('tags' not in file_metadata or
+       'description' not in file_metadata):
+        return False
+    if('creation_date'in file_metadata and not
+       is_valid_datetime(file_metadata["creation_date"])):
+        return False
+    return True
```

Comment:
Since I found this, you might want to re-use it! (To add more validation.) Sorry I hadn't dug it up earlier. https://github.com/OpenHumans/open-humans/blob/master/private_sharing/forms.py#L238